CMakeLists.txt

```diff
@@ -54,6 +54,12 @@ option(TEST_INTEL_OPTIMIZATION "Test intel optimization and vectorization" OFF)
 option(TEST_ERROR_CHECKING "Enables the error checking unit test. This test will FAIL" OFF)
 option(CODE_COVERAGE "Enable code coverage reporting" OFF)
 
+# HDF5 compression level option, value between 0 and 9
+set(PARTHENON_HDF5_COMPRESSION_LEVEL 5 CACHE STRING "Compression level for HDF5 output and restart files, valid range is 0-9, default is 5, set to 0 to turn off compression")
+if(NOT PARTHENON_HDF5_COMPRESSION_LEVEL MATCHES "^[0-9]$")
+  message(FATAL_ERROR "PARTHENON_HDF5_COMPRESSION_LEVEL must be a number between 0 and 9")
+endif()
```

Comment:
The compression does not need to be set at compile time. It can be set at run-time (in fact, it could be changed during runtime). The reason I made it a compile time option is so that we can make sure the version of HDF5 used doesn't have the known bug. However, I could make that option simply enable/disable compression but not set the compression level and then we could set the compression level through the input file (individually for each HDF5 output/restart file). If one tries to use compression with a build that has compression disabled, we can generate an error or warning.

I think that's actually a pretty good idea. I'll go ahead and make that change.